include/GPM/Random.inl

```diff
@@ -0,0 +1,154 @@
+#include "Random.hpp"
+
+void initSeed()
+{
+    std::srand(time(NULL));
+}
+
+void initSeed(float seed)
+{
+    srand(seed);
+}
+
+template<typename T = float> 
+auto unitValue() -> std::enable_if_t<std::is_floating_point<T>::value, T>
+{
+    return static_cast <T> (rand()) / static_cast <T> (RAND_MAX);
+} 
+
+template<typename T = int> 
+auto unitValue() -> std::enable_if_t<std::is_integral<T>::value, T>
+{
+    return static_cast<T>(rand() % static_cast<T>(2));
+}
+
+template<typename T = float>
+auto ranged(const T& max) -> std::enable_if_t<std::is_floating_point<T>::value, T>
+{
+    return max <= std::numeric_limits<T>::epsilon() ? static_cast<T>(0) : static_cast <T> (rand()) / (static_cast <T> (RAND_MAX / max));
+}
+
+template<typename T = int>
+auto ranged(const T& max) -> std::enable_if_t<std::is_integral<T>::value, T>
+{
+    return max <= std::numeric_limits<T>::epsilon() ? static_cast<T>(0) : ranged<T>(0, max);
+}
+
+template<typename T>
+auto ranged(const T& min, const T& max)  -> std::enable_if_t<std::is_floating_point<T>::value, T>
+{
+    return max - min <= std::numeric_limits<T>::epsilon() ? max : min + static_cast <T> (rand()) /( static_cast <T> (RAND_MAX/(max - min)));
+}
+
+template<typename T>
+auto ranged(const T& min, const T& max) -> std::enable_if_t<std::is_integral<T>::value, T>
+{
+    return max - min <= std::numeric_limits<T>::epsilon() ? max : min + static_cast <T> (rand()) / (static_cast <T> (RAND_MAX/((max + static_cast<T>(1)) - min)));
+}
```

Comment:
The branches in these functions will probably cost more than doing the actual calculation, even when `max - min <= std::numeric_limits<T>::epsilon()`. I suggest removing the branches, and leaving only the second part of the ternary operations